types/duration.go

```diff
@@ -0,0 +1,295 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+	"unicode"
+
+	"github.com/cedar-policy/cedar-go/internal/consts"
+)
+
+var unitToMillis = map[string]int64{
+	"d":  consts.MillisPerDay,
+	"h":  consts.MillisPerHour,
+	"m":  consts.MillisPerMinute,
+	"s":  consts.MillisPerSecond,
+	"ms": 1,
+}
+
+var unitOrder = []string{"d", "h", "m", "s", "ms"}
+
+// A Duration is a value representing a span of time in milliseconds.
+type Duration struct {
+	Value int64
```

Comment:
I think we should strongly consider making `Value` private. The accessor methods with units seem like the right public interface for this.

Reply:
Changed. Note that Decimal still exposes this.